lib/remote_storage/riak.rb

```diff
@@ -68,7 +68,8 @@ def get_data(user, directory, key)
       when "application/json"
         return object.data.to_json
       else
-        return serializer_for(object.content_type) ? object.data : object.raw_data
+        # Never return nil, always turn data into a string
```

Comment:
Not sure how strings work in Ruby, but if it's the same as in JavaScript, then turning the raw data into a string would corrupt binary data where not all bytes can necessarily be interpreted as utf-8 characters.